table/table_builder.h

```diff
@@ -56,7 +56,8 @@ struct TableBuilderOptions {
       CompressionType _compression_type,
       const CompressionOptions& _compression_opts,
       const std::string* _compression_dict, bool _skip_filters,
-      const std::string& _column_family_name, int _level)
+      const std::string& _column_family_name, int _level,
+      const uint64_t _creation_time = 0)
```

Comment:
I am a little hesitant to force creation_time to be provided always, as, I believe, there is no other usage for it at the moment other than in FIFO-compaction-with-TTL. I need to spend some more time thinking about it, and will defer adding it to a later PR.